### PR TITLE
Update the font_table example.

### PR DESCRIPTION
**Pull request by anntzer**
Print characters corresponding to codepoints >0xff.
Add comments explaining font charmaps (briefly).
Don't use pyplot.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
This seems OK, but why did you change the name of the file? That will break links to the example...

**Comment by jklymak** [comment]
https://4997-7439715-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/gallery/text_labels_and_annotations/font_table_sgskip.html#sphx-glr-gallery-text-labels-and-annotations-font-table-sgskip-py 

Is that how you wanted it to look?  `Usage::` doesn't seem to work...  Is there to be nothing shown?

**Comment by anntzer** [comment]
I think an empty line after `Usage::` will fix this, let's see.

The file is sgskipped because sphinx-gallery can't handle stuff like this (which tries to read from sys.argv) properly, so nothing is rendered in the docs.

File: examples/text_labels_and_annotations/font_table_sgskip.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+==========
+Font table
+==========
+
+Matplotlib's font support is provided by the FreeType library.
+
+Here, we use `~.Axes.table` build a font table that shows the glyphs by Unicode
+codepoint, and print the glyphs corresponding to codepoints beyond 0xff.
+
+Usage::
```
Comment (timhoffm) [inline examples/text_labels_and_annotations/font_table_sgskip.py]:
```suggestion
You can download this script and run it to investigate a font by calling::
```

File: examples/text_labels_and_annotations/font_table_sgskip.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    python font_table_sgskip.py /path/to/font/file
+"""
+
+import unicodedata
+
+from matplotlib import (
```
Comment (timhoffm) [inline examples/text_labels_and_annotations/font_table_sgskip.py]: Is that really common style? I would have used
~~~
import matplotlib.pyplot as plt
import matplotlib.font_manager as fm
~~~

File: examples/text_labels_and_annotations/font_table_sgskip.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    font_manager as fm,
+    pyplot as plt,
+)
+from matplotlib.ft2font import FT2Font
+import numpy as np
+
+
+def main(path):
```
Comment (timhoffm) [inline examples/text_labels_and_annotations/font_table_sgskip.py]:
```suggestion
def draw_font_table(path):
```
While it's technically a `main()` the semantic name tells us what the function is actually doing.

**Comment by anntzer** [comment]
> What's the motivation to make this file an executable script? Should users download the file and actually use it as a standalone tool? If so, the sgskip title is really bad.

Or just run it from the examples folder if they already have a cloned repo.
What else do you suggest other than sgskip?

> Is it possible to add a check to see if this was run from the command line or by sphinx galery and run draw_font_table(None) in the latter case?

Likely, but my interest in figuring out how to make s-g play nicely with this is close to non-existent.

File: examples/text_labels_and_annotations/font_table_sgskip.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+    python font_table_sgskip.py /path/to/font/file
+"""
+
+import unicodedata
+
+import matplotlib.font_manager as fm
+from matplotlib.ft2font import FT2Font
+import matplotlib.pyplot as plt
+import numpy as np
```
Comment (timhoffm) [inline examples/text_labels_and_annotations/font_table_sgskip.py]: unused import

**Comment by tacaswell** [comment]
I have a mild preference to not rename examples to avoid changing the URLs of the example but will not block merging over that.

**Comment by timhoffm** [comment]
Actually, you don't have to do much to show the table. I just removed the `_sgskip` part and the table for the default font got rendered.

One should consider making the `print()` opt-in. Otherwise there's a long output box in the rendered example.

image

I could push the suitable changes if desired.

**Comment by anntzer** [comment]
If s-g can run this then yes, let's include the table.  (In which case we're going to get a rename anyways.)
Seems reasonable to make the print() opt-in (e.g. with a `--print-non-latin1` option).
Please push the changes if you already have them ready.

@tacaswell We should really redirect `matplotlib.org/...` to `matplotlib.org/3.$x.$y/...` so that people only bookmark versioned links... (basically the same strategy as cpython.)

**Comment by timhoffm** [comment]
I've added the inclusion of the table in the sphinx-gallery output as a separte PR.

Should be squashed before merging.

**Comment by timhoffm** [comment]
I've additionally factored out the printing of the glyphs. This makes the table code more readable (and at least part of the motivation of the example is showing how to use `table()`).

File: examples/text_labels_and_annotations/font_table.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+import unicodedata
+
+import matplotlib.font_manager as fm
+from matplotlib.ft2font import FT2Font
+import matplotlib.pyplot as plt
+
+
+def print_glyphs(font):
+    """Print the all the glyphs in the given FT2Font font to stdout."""
```
Comment (anntzer) [inline examples/text_labels_and_annotations/font_table.py]: remove "the"

File: examples/text_labels_and_annotations/font_table.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    print("The font face contains the following glyphs:")
+    for char_code, glyph_index in charmap.items():
+        char = chr(char_code)
+        name = unicodedata.name(
+                char,
+                f"{char_code:#x} ({font.get_glyph_name(glyph_index)})")
+        print(f"{glyph_index:>{max_indices_len}} {char} {name}")
+
+
+def draw_font_table(path, print_all=False):
```
Comment (anntzer) [inline examples/text_labels_and_annotations/font_table.py]: Make print_all kwonly?  (Not that it really matters here but it's the kind of arg that should likely be kwonly if this was a general-use function...)

**Comment by timhoffm** [comment]
This is how it looks like

https://5043-7439715-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/gallery/text_labels_and_annotations/font_table.html

**Pull request merged**